workspace/workspace.go

```diff
@@ -298,3 +303,59 @@ func validWorkspaceConfig(workspace Workspace) error {
 
 	return nil
 }
+
+// findWorkspaceRoot looks in the directory above the manifest file, if there's a config.yml, use that
+// otherwise we use the directory of the manifest file as the workspace root
+func findWorkspaceRoot() (string, error) {
+	wd, err := os.Getwd()
+
+	if err != nil {
+		panic(err)
+	}
+
+	if _, err := os.Stat(filepath.Join(wd, "config.yml")); err == nil {
+		// If we're currently in the directory with the config.yml
+		return wd, nil
+	}
+
+	// Look upwards to find a manifest file
+	packageDir, err := findNearestManifestFile()
+	if err != nil {
+		return "", err
+	}
+
+	// If we find a manifest file, check the parent directory for a config.yml
+	parent := filepath.Dir(packageDir)
+	if _, err := os.Stat(filepath.Join(parent, "config.yml")); err == nil {
+		return parent, nil
+	} else {
+		return packageDir, nil
+	}
+
+	// No config in the parent of the package? No workspace!
+	return "", fmt.Errorf("searching for a workspace root")
+}
+
+func findFileUpTree(filename string) (string, error) {
+	wd, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+
+	for {
+		file_path := filepath.Join(wd, filename)
+		if _, err := os.Stat(file_path); err == nil {
+			return wd, nil
+		}
+
+		wd = filepath.Dir(wd)
+
+		if strings.HasSuffix(wd, "/") {
```

Comment:
This tripped me up a bit until I researched it, can we add a comment that only root will ever have a trailing `/`?